src/coreclr/nativeaot/System.Private.CoreLib/src/System/AppContext.NativeAot.cs

```diff
@@ -36,5 +37,21 @@ internal static void OnUnhandledException(object e)
         {
             UnhandledException?.Invoke(/* AppDomain */ null, new UnhandledExceptionEventArgs(e, true));
         }
+
+        private static unsafe string GetRuntimeModulePath()
+        {
+            // We aren't going to call this method, we just need an address that we know is in this module.
+            // As this code is NativeAOT only, we know that this method will be AOT compiled into the executable,
+            // so the entry point address will be in the module.
+            void* ip = (void*)(delegate*<string>)&GetRuntimeModulePath;
+            if (RuntimeAugments.TryGetFullPathToApplicationModule((nint)ip, out _) is string modulePath)
```

Comment:
Can we mark `TryGetFullPathToApplicationModule` with a `string?` return? Right now it is only `string` and it took me a moment to realize what this `if` was actually checking.